openedx_external_enrollments/external_enrollments/pathstream_external_enrollment.py

```diff
@@ -0,0 +1,65 @@
+"""PathstreamExternalEnrollment class file."""
+import logging
+from datetime import datetime
+
+from openedx_external_enrollments.external_enrollments.base_external_enrollment import BaseExternalEnrollment
+from openedx_external_enrollments.models import ExternalEnrollment
+
+LOG = logging.getLogger(__name__)
+
+
+class PathstreamExternalEnrollment(BaseExternalEnrollment):
+    """
+    PathstreamExternalEnrollment class.
+    """
+    def __str__(self):
+        return 'pathstream'
+
+    def _get_enrollment_data(self, data, course_settings):
+        """
+        Returns a string with the data required to be treated as a log.
+        String format: 'course_key,email,date_time,status'
+        """
+        enrollment_data = u'{course_key},{email},{date_time},{status}\n'.format(
+            course_key=data.get('course_id'),
+            email=data.get('user_email'),
+            date_time=datetime.utcnow(),
+            status=str(data.get('is_active')).lower(),
+        )
+        return enrollment_data
+
+    def _post_enrollment(self, data, course_settings=None):
+        """
+        Save enrollment data in ExternalEnrollment model.
+        """
+        LOG.info('calling enrollment for [%s] with data: %s', self.__str__(), data)
+        LOG.info('calling enrollment for [%s] with course settings: %s', self.__str__(), course_settings)
+        details = {
+            'enrollment_data_formated': self._get_enrollment_data(data, course_settings),
+            'is_uploaded': False,
+        }
+        enrollment, created = ExternalEnrollment.objects.get_or_create(  # pylint: disable=no-member
+            controller_name=str(self),
+            course_shell_id=data.get('course_id'),
+            email=data.get('user_email'),
+        )
+
+        if created:
+            enrollment.meta = [
+                details,
+            ]
+        else:
+            enrollment.meta.append(details)
```

Comment:
This could be an inline if assignment. @Jacatove

Reply:
Inline if assignment returns None when doing the append to the variable.
![image](https://user-images.githubusercontent.com/40271196/126525324-cd4d7849-f12d-4663-aaec-832dbf5cb1a7.png)
@Squirrel18